src/Microsoft.AspNetCore.Razor.Language/Intermediate/RazorIRNodeExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Razor.Language.Intermediate
+{
+    public static class RazorIRNodeExtensions
+    {
+        private static readonly IReadOnlyList<RazorDiagnostic> EmptyDiagnostics = new RazorDiagnostic[0];
+
+        public static IReadOnlyList<RazorDiagnostic> GetAllDiagnostics(this RazorIRNode node)
```

Comment:
No, I don't like ISet for this. It's mutable and has lots of 'extras' on the API